src/test/java/org/kiwiproject/registry/consul/client/ConsulRegistryClientTest.java

```diff
@@ -0,0 +1,129 @@
+package org.kiwiproject.registry.consul.client;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.net.HostAndPort;
+import com.orbitz.consul.Consul;
+import com.orbitz.consul.model.agent.ImmutableRegistration;
+import com.pszymczyk.consul.junit.ConsulExtension;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.kiwiproject.registry.client.RegistryClient;
+import org.kiwiproject.registry.consul.config.ConsulConfig;
+
+@DisplayName("ConsulRegistryClient")
+class ConsulRegistryClientTest {
+
+    // NOTE: Even though this extension uses an AfterAllCallback, it can NOT be static as running all of the tests fail. I'm not sure if this is
+    //       something with the extension or with the Nested test classes
+    @RegisterExtension
+    ConsulExtension CONSUL = new ConsulExtension();
```

Comment:
You should be able to make this `final`

Reply:
I might also suggest renaming to `consulExtension` because:

(a) There is a `consul` var in `setUp` of type `Consul` and this is not a `Consul` so it could maybe possibly be a tad confusing
(b) it is not static, so maybe we should not name it like it is `static final` in `ALL_CAPS`, and instead just name it like a normal instance field.

I'm not sure how consistent we are about this, however, across our libraries.